Tribler/Core/TorrentChecker/session.py

```diff
@@ -198,309 +185,384 @@ def __init__(self, tracker_url, tracker_address, announce_page, on_result_callba
 
     def max_retries(self):
+        """
+        Returns the max amount of retries allowed for this session.
+        :return: The maximum amount of retries.
+        """
         return HTTP_TRACKER_MAX_RETRIES
 
     def retry_interval(self):
+        """
+        Returns the interval one has to wait before retrying to connect.
+        :return: The interval before retrying.
+        """
         return HTTP_TRACKER_RECHECK_INTERVAL
 
     def create_connection(self):
-        self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        self._socket.setblocking(0)
-
-        return self.recreate_connection()
-
-    def recreate_connection(self):
-        super(HttpTrackerSession, self).recreate_connection()
-
-        # an exception may be raised if the socket is non-blocking
-        try:
-            self._socket.connect(self._tracker_address)
-        except Exception as e:
-            # Error 115 means the operation is in progress.
-            # Error 10035 is WSAEWOULDBLOCK on Windows.
-            if e[0] not in (115, 10035):
-                self._logger.debug(u"%s Failed to connect to HTTP tracker [%s, %s]: %s",
-                                   self, self._tracker_url, self._tracker_address, e)
-                self._is_failed = True
-                return False
-
+        super(HttpTrackerSession, self).create_connection()
         self._action = TRACKER_ACTION_CONNECT
-        self._last_contact = int(time.time())
         return True
 
-    def _handle_connection(self):
+    def connect_to_tracker(self):
         # create the HTTP GET message
         # Note: some trackers have strange URLs, e.g.,
         #       http://moviezone.ws/announce.php?passkey=8ae51c4b47d3e7d0774a720fa511cc2a
-        #       which has some sort of 'key' as parameter, so we need to check
-        #       if there is already a parameter available
-        message = 'GET '
-        message += '/' + self._announce_page.replace(u'announce', u'scrape')
-        if message.find('?') == -1:
-            message += '?'
-        else:
-            message += '&'
+        #       which has some sort of 'key' as parameter, so we need to use the add_url_params
+        #       utility function to handle such cases.
 
-        # append the infohashes as parameters
-        for infohash in self._infohash_list:
-            message += 'info_hash='
-            message += urllib.quote(infohash)
-            message += '&'
-        message = message[:-1]  # remove the last AND '&'
-        message += ' HTTP/1.1\r\n'
-        message += '\r\n'
-
-        try:
-            self._socket.sendall(message)
-        except Exception as e:
-            self._logger.info(u"%s Failed to send HTTP SCRAPE message: %s", self, e)
-            self._is_failed = True
-            return
+        url = add_url_params("http://%s:%s/%s" % (self._tracker_address[0], self._tracker_address[1], self._announce_page.replace(u'announce', u'scrape')),
```

Comment:
don't go over 120 if you can avoid it. (a while ago, I pushed a pylintrc config file you can use to automatically check for this kind of things).